app/main.py

```diff
@@ -1,13 +1,26 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "name": error.get("code"),
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "errors": [format_linter_error(err_point) for err_point in errors],
+        "path": file_path,
+        "status": "failed" if bool(errors) else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+
+    return [
+        format_single_linter_file(key, linter_report[key])
+        for key in linter_report.keys()
```

Comment:
I case when you are operating with both key and value of dict items it better to iterate over key and value. It will be more clear that you use both values